File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n,m = len(matrix),len(matrix[0])
        x,y = n-1,0

        while x>=0 and y<m:
            if matrix[x][y] == target:
                return True
            elif matrix[x][y] > target: x-=1
            else: y+=1
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const visited = {};

        for (let i=0; i<nums.length; i++) {
            if (visited.hasOwnProperty(target-nums[i])) {
                return [visited[target-nums[i]], i];
            }
            visited[nums[i]] = i;
        }
        return [-1,-1];
    }
}
